### ISSUE TITLE
[restrict-plus-operands] Doesn't check compound assignment

### ISSUE THREAD

**Issue by phaux** [typescript-eslint/typescript-eslint#878]
**Repro**
```JSON
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": "error"
  }
}
```
```TS
declare let buffer: string | undefined
buffer += "some data"
```

**Expected Result**
Should be an error.

**Actual Result**
No error.

**More info**
`buffer = buffer + "data"` is correctly reported as error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0` |
| `@typescript-eslint/parser`        | `2.0.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |
